test_data/asm/block_to_block_empty_submachine.asm

```diff
@@ -0,0 +1,54 @@
+// This test is mostly a copy of block_to_block.asm, but:
+// - The machines have dynamic length (either 4 or 8)
+// - `instr_add` can be set by the prover.
+//
+// We set `instr_add` to zero in all rows at runtime. This simulates
+// a secondary machine that is never called (and, because it depends on
+// runtime information, it's not removed by the optimizer).
+
+machine Arith with
+    latch: latch,
+    operation_id: operation_id
+{
+    operation add<0> x, y -> z;
+
+    col fixed operation_id = [0]*;
+    col fixed latch = [1]*;
+    col witness x;
+    col witness y;
+    col witness z;
+    z = x + y;
+}
+
+machine Main with
+    min_degree: 4,
+    max_degree: 8,
+    latch: latch,
+    operation_id: operation_id
+{
+    Arith arith(4, 8);
+
+    // return `x + y`, where the prover can chose whether to add locally or externally
```

Comment:
And here it would also be `3*x + 3*y` for the operation (for each block)

Reply:
Right!